lib/profile/carbs.js

```diff
@@ -16,11 +16,26 @@ function carbRatioLookup (inputs) {
                     break;
                 }
             }
+            return carbRatio.ratio;
+        } else {
+            if (carbratio_data.units == "exchanges") {
+            //carbratio_data.schedule.sort(function (a, b) { return a.offset > b.offset });
+            carbRatio = carbratio_data.schedule[carbratio_data.schedule.length - 1];
+
+            for (var i = 0; i < carbratio_data.schedule.length - 1; i++) {
+                if ((now >= getTime(carbratio_data.schedule[i].offset)) && (now < getTime(carbratio_data.schedule[i + 1].offset))) {
+                    carbRatio = carbratio_data.schedule[i];
+                    break;
+                }
+            }
+            carbRatio.ratio = 12 / carbRatio.ratio
```

Comment:
We could refactor this down to eliminate all the duplicate code (by running it if (carbratio_data.units == "grams" ||(carbratio_data.units == "exchange")) and then splitting out the return lines based on grams or exchanges.  Do you want to take a shot at that, or should I merge this and then we can refactor later?